### PR TITLE
Fix contour plots by respecting matplotlib path codes

### PR DESCRIPTION
**Pull request by LDAP**
tikzplotlib did not respect path codes like `MOVETO` or `LINETO`. For details refer to [Matplotlib Docs](https://matplotlib.org/stable/api/path_api.html).

Before the fix:
grafik

After the fix:
grafik

The issues were in this `addplot`:
```latex
\addplot [draw=none, fill opacity=0.7, fill=color4]
table{%
x  y
0.25 0.110283316044848
0.5 0.162226841253728
0.681096466688732 0.25
0.75 0.324252010875173
1 0.397922874111693
1 0.437026993464228
0.75 0.47199448612388
0.698026588478406 0.5
0.75 0.515385342139691
1 0.578468346553991
1 0.627194550968263
0.75 0.596550377722359
0.639506844367552 0.75
0.5 0.858709673639598
0.346569627641532 0.75
0.477127362265474 0.5
0.25 0.389206122580593
0 0.267768369278118
0 0.25
0 0.232095584068258
0.25 0.110283316044848

0.0564486443589441 0.25
0.25 0.349037213178555
0.5 0.32690805798404
0.543996091294153 0.25
0.5 0.22867613887788
0.25 0.150599551535844
0.0564486443589441 0.25

0.410846444385108 0.75
0.5 0.813167766497112
0.581063032154561 0.75
0.5 0.585603136270736
0.410846444385108 0.75
};
```

without the newlines `pgfplots` produces the wrong result.

### CODE REVIEW

**Comment by LDAP** [comment]
I checked the failing tests. Besides a higher precision in the most current version the reference lacks the same new lines as in my example. The issue was not detected because in every case there were only two parts instead of three which are needed to form a area.

Reference:
```latex
\addplot [draw=none, fill=color248]
table{%
x  y
0.968 1
1 1.032
1.032 1
1 0.968
0.968 1
1 0.984
1.016 1
1 1.016
0.984 1
1 0.984
};
```

Fixed:
```latex
\addplot [draw=none, fill=color248]
table{%
x  y
0.968 1
1 1.032
1.032 1
1 0.968
0.968 1

1 0.984
1.016 1
1 1.016
0.984 1
1 0.984
};
```

I did update the tests with the higher precision and the needed newlines for a correct representation of the `MOVETO` segments.

**Comment by LDAP** [comment]
For an explanation why an empty line fixes the issue please refer to [PGFPlots Manual](https://ctan.kako-dev.de/graphics/pgf/contrib/pgfplots/doc/pgfplots.pdf#pgfp./pgfplots/empty%3Aline)

File: tikzplotlib/_path.py [hunk 1]
```diff
@@ -250,7 +250,20 @@ def draw_pathcollection(data, obj):
...
+            # https://matplotlib.org/stable/api/path_api.html
+            codes = (
+                path.codes
+                if path.codes is not None
+                else np.array([1] + [2] * (len(dd) - 1))
+            )
+            dd_strings = []
+            for row, code in zip(dd, codes):
+                if code == 1:  # MOVETO
+                    dd_strings += [
```
Comment (nschloe) [inline tikzplotlib/_path.py:262]: How about the shorter `.append([])` here?

Comment (LDAP) [reply tikzplotlib/_path.py:262]: Sure. I have no preferences there :)

Comment (LDAP) [reply tikzplotlib/_path.py:262]: Actually, I can move the comment upwards to make it consistent with the else branch?

**Comment by nschloe** [comment]
Thanks!

**Pull request merged**